pulsar-client-tools/src/main/java/org/apache/pulsar/client/cli/CmdConsume.java

```diff
@@ -171,6 +174,12 @@ private String interpretMessage(Message<?> message, boolean displayHex) throws I
         } else if (value instanceof GenericRecord) {
             Map<String, Object> asMap = genericRecordToMap((GenericRecord) value);
             data = asMap.toString();
+        } else if (value instanceof ByteBuffer) {
+            ByteBuffer payload = ((ByteBuffer)value);
```

Comment:
what about using the internal array of the ByteBuffer in case of "hasArray", offset=0 and len == remaining? I did the same in BytesKafkaSource
this way we can save a copy.